models/slice/matplotlib_slice_plotter.py

```diff
@@ -1,25 +1,97 @@
 from slice_plotter import SlicePlotter
-from mantid.simpleapi import AnalysisDataService
+from presenters.slice_plotter_presenter import INVALID_INTENSITY,INVALID_PARAMS,INVALID_SMOOTHING,INVALID_X_PARAMS,\
+    INVALID_X_UNITS,INVALID_Y_PARAMS,INVALID_Y_UNITS
+from models.workspacemanager.mantid_workspace_provider import MantidWorkspaceProvider
+from mantid.simpleapi import BinMD
+from mantid.api import IMDEventWorkspace
+from math import floor
+import numpy as np
+import matplotlib.cm as cm
+
 from plotting import pyplot as plt
 
 
 def get_aspect_ratio(workspace):
     return 'auto'
 
+def to_float(x):
+    if x:
+        return float(x)
+    else:
+        return None
 
 class MatplotlibSlicePlotter(SlicePlotter):
+    def __init__(self):
+        self._workspace_provider = MantidWorkspaceProvider()
+        self._colormaps = ['viridis', 'jet', 'summer', 'winter']
+
     def display_slice(self, selected_workspace, x_axis, y_axis, smoothing, intensity_start, intensity_end, norm_to_one,
```

Comment:
I think having a `SlicePlotter` with a `display_slice` is a good design. However this is not just `displaying` the slice it's doing a whole load of computation including manipulating the inputs. I think manipulation of inputs should be done in the presenter or some auxillary method/class in that testable zone.

How about following the `WorkspaceProvider` approach and having a `SlicingAlgorithm` of which your `MantidSlicingAlgorithm` actually runs `BinMD`?